### ISSUE TITLE
fluxctl does not work with Digital Ocean auth

### ISSUE THREAD

**Issue by moltar** [weaveworks/flux#1959]
KUBECONFIG:

```yaml
users:
- name: do-nyc1-foo-admin
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - kubernetes
      - cluster
      - kubeconfig
      - exec-credential
      - --version=v1beta1
      - UUID
      command: doctl
      env: null
```

fluxctl output:

```
Error: (kubeconfig.exec-credential) command is missing required arguments
Error: Could not create a dialer: Could not get pod name: Listing pods in kubernetes: Get https://UUID.k8s.ondigitalocean.com/api/v1/namespaces/flux/pods?labelSelector=app%3Dflux: getting credentials: exec: exit status 1
Run 'fluxctl list-workloads --help' for usage.
```

Thanks!

**Comment by 2opremio** [comment]
This means that `doctl` exits with non-zero status, indicating it failed. Does it work when you do `kubectl get pods --selector=app=flux`?

What version of `fluxctl` is this? 

Can you show the (redacted) output of `doctl kubernetes cluster kubeconfig exec-credential --version=v1beta1 UUID; echo $?`?

**Comment by moltar** [comment]
```
❯ kubectl get pods --selector=app=flux
No resources found.

❯ kubectl --namespace=flux get pods --selector=app=flux
No resources found.

❯ kubectl --namespace=flux get pods
NAME                         READY   STATUS    RESTARTS   AGE
flux-b555bfdf5-2xq4k         1/1     Running   0          79m
memcached-549dd79996-fppn2   1/1     Running   0          102m
```

---

```
❯ doctl kubernetes cluster kubeconfig exec-credential --version=v1beta1 UUID; echo $?
{"kind":"ExecCredential","apiVersion":"client.authentication.k8s.io/v1beta1","spec":{},"status":{"expirationTimestamp":"2019-04-26T08:50:24Z","clientCertificateData":"...."}}
0
```

**Comment by 2opremio** [comment]
What version of fluxctl are you running?

**Comment by moltar** [comment]
Sorry, not at the computer now. But I upgraded it today via brew. So whatever brew latest is. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by 2opremio** [comment]
If posssible, please confirm the version when you are infront of the computer.

**Comment by moltar** [comment]
```
❯ fluxctl version
1.12.0
```

**Comment by 2opremio** [comment]
Thanks. I will install a DO cluster and try to reproduce.

**Comment by 2opremio** [comment]
@moltar would you mind make a last test for me? Can you try the `master` binary of flux? We recently bumped the kubernetes client version (#1929 ), which may have fixed this problem.

Here is the binary: [fluxctl.gz](https://github.com/weaveworks/flux/files/3107200/fluxctl.gz)

**Comment by moltar** [comment]
Yup, great success!

```
❯ ./fluxctl --k8s-fwd-ns="flux" identity
ssh-rsa AAAA...
```

Thank you!

**Comment by 2opremio** [comment]
Great! This means it will be fixed in the next release including #1929
